lightwood/api/data_source.py

```diff
@@ -181,11 +181,13 @@ def __getitem__(self, idx):
                 col_config = self.get_column_config(col_name)
 
                 if col_name not in self.encoded_cache:  # if data is not encoded yet, encode values
-                    if not ((dropout_features is not None and col_name in dropout_features) or not self.enable_cache):
+                    if not ((dropout_features is not None and col_name in dropout_features) or not self.enable_cache or
+                            'previous_' in col_name):  # TODO: fragile check for name
```

Comment:
check should be:

`col_name.startswith('previous_') and col_name in self.outpout_features`

Reply:
Second clause seems off, `previous_` columns are input features.

How about this check: `col_name.startswith('previous_') and feature_set == 'input_features'`?

Reply:
Yes, but it's `previous_{name}` where `name` is always the name of an output feature.

Reply:
Ah, I get what you meant now